base/settings/tests/testthat/test.clean.settings.R

```diff
@@ -0,0 +1,35 @@
+PEcAn.logger::logger.setQuitOnSevere(FALSE)
+on.exit(PEcAn.logger::logger.setQuitOnSevere(TRUE))
+
+test_that("`test.clean.settings` works correctly for invalid and correct inputs", {
+
+  # Error if input file is NULL or does not exist
+  expect_error(
+    clean.settings(inputfile = NULL), 
+    "Could not find input file."
+  )
+  expect_error(
+    clean.settings(inputfile = "nonexistent.xml"), 
+    "Could not find input file."
+  )
+
+  # Works if correct input file provided
+  t <- withr::with_tempfile("tf", {
+    clean.settings(inputfile = "data/testinputcleanup.xml", outputfile = tf)
+    readLines(tf)
+  })
+
+  t <- XML::xmlToList(XML::xmlParse(t))
+	
+  # Check for updated settings after cleanup
+  expect_equal(t$outdir, "pecan")
+  expect_equal(t$rundir, NULL)
+  expect_equal(t$modeloutdir, NULL)
+  expect_equal(t$host$rundir, NULL)
+  expect_equal(t$host$outdir, NULL)
+  expect_equal(t$database$dbfiles, NULL)
+  expect_equal(t$workflow, NULL)
+  expect_equal(t$pfts[[1]]$pft$outdir, NULL)
+  expect_equal(t$pfts[[1]]$pft$posteriorid, NULL)
+  expect_equal(t$host, list(name = "localhost"))
```

Comment:
What you have here works already, but I think (but didn't test before suggesting 😄) that you could further isolate tests from unwanted side effects by move all the expectations inside the `with_tempfile` call. Then `t` is scoped to just this test and can't carry over into other ones.

```suggestion
  withr::with_tempfile("tf", {
    clean.settings(inputfile = "data/testinputcleanup.xml", outputfile = tf)
    test_xml <- readLines(tf)
    
    t <- XML::xmlToList(XML::xmlParse(test_xml))
	
    # Check for updated settings after cleanup
    expect_equal(t$outdir, "pecan")
    expect_equal(t$rundir, NULL)
    expect_equal(t$modeloutdir, NULL)
    expect_equal(t$host$rundir, NULL)
    expect_equal(t$host$outdir, NULL)
    expect_equal(t$database$dbfiles, NULL)
    expect_equal(t$workflow, NULL)
    expect_equal(t$pfts[[1]]$pft$outdir, NULL)
    expect_equal(t$pfts[[1]]$pft$posteriorid, NULL)
    expect_equal(t$host, list(name = "localhost"))
  })

```